docs/csharp/tutorials/inheritance.md

```diff
@@ -0,0 +1,294 @@
+
+## Prerequisites ##
+
+This tutorial assumes that you've installed .NET Core. For installation instructions, see [.NET Core installation guide](https://www.microsoft.com/net/core). You also need a code editor. This tutorial uses [Visual Studio Code](https://code.visualstudio.com), although you can use any code editor of your choice.
+
+## Running the examples ##
+
+To create and run the examples in this tutorial, you use the [dotnet](../../core/tools/dotnet.md) utility from the command line. Follow these steps for each example:
+
+1. Create a directory to store the example.
+
+1. Enter the [dotnet new](../../core/tools/dotnet-new.md) command from the command line to create a new .NET Core project.
+
+1. Copy and paste the code from the example into your code editor.
+
+1. Enter the [dotnet restore](../../core/tools/dotnet-restore.md) command from the command line to load or restore the project's dependencies.
+
+1. Enter the [dotnet run](../../core/tools/dotnet-run.md) command to compile and execute the example.
+
+## Background: What is inheritance? ##
+
+*Inheritance* is one of the fundamental attributes of object-oriented programming. It allows you to define a child class that reuses (inherits), extends, or modifies the behavior of a parent class. The class whose members are inherited is called the *base class*. The class that inherits the members of the base class is called the *derived class*.
+
+C# and .NET support *single inheritance* only. That is, a class can only inherit from a single class. However, inheritance is transitive, which allows you to define an inheritance hierarchy for a set of types. In other words, type `D` can inherit from type `C`, which inherits from type `B`, which inherits from the base class type `A`. Because inheritance is transitive, the members of type `A` are availble to type `D`.
+
+Not all members of a base class are inherited by derived classes. The following members are not inherited:
+
+- [Constructors](../programming-guide/classes-and-structs/constructors.md), which are called to create a new instance of the class. Each class must define its own constructors.
+
+- [Destructors](../programming-guide/classes-and-structs/destructors.md), which are called by the runtime's garbage collector to destroy instances of a class.
+
+- [Static members](../programming-guide/classes-and-structs/static-classes-and-static-class-members.md). Only instance members are inherited by base classes.
+
+- [Private](../language-reference/keywords/private.md) and [internal](../language-reference/keywords/internal.md) members. However, internal members of a base class may be *visible* to a derived class if the two types are in the same assembly.
```

Comment:
Internal members are inherited.  Their visibility is limited to the assembly  where the base class is declared, but they are inherited.